weave/tests/test_artifact_metadata.py

```diff
@@ -93,3 +93,26 @@ def test_artifact_metadata(user_by_api_key_in_env):
         "k_3": "v_4",
         "k_4": "v_6",
     }
+
+
+def test_artifact_files_count(user_by_api_key_in_env):
+    run = wandb.init(project="project_exists")
+    artifact = wandb.Artifact("test", "datatest")
+    table1 = wandb.Table(data=[[1, 2, 3]], columns=["a", "b", "c"])
+    table2 = wandb.Table(data=[[1, 2, 3]], columns=["a", "b", "c"])
+    table3 = wandb.Table(data=[[1, 2, 3]], columns=["a", "b", "c"])
+    artifact.add(table1, "table1")
+    artifact.add(table2, "table2")
+    artifact.add(table3, "table3")
```

Comment:
can you please add a table at `some/nested/dir/table4` to ensure we count nested files